### PR TITLE
Show .editorconfig errors in frontend

### PR DESCRIPTION
**Pull request by JakobDev**
If the user views the .editorconfig of the Repo, an the the .editorconfig contains error, those errors are now shown above the file.

### CODE REVIEW

**Comment by silverwind** [comment]
Got a screenshot?

**Comment by JakobDev** [comment]
> Got a screenshot?

Sure

image

This error box could also be used for other files (e.g. Issue/PR templates) in the future.

**Comment by silverwind** [comment]
Yes, could be used elsewhere. Looks great.

**Review by silverwind — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

File: routers/web/repo/view.go [hunk 1]
```diff
@@ -374,6 +374,11 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 	ctx.Data["FileName"] = blob.Name()
 	ctx.Data["RawFileLink"] = rawLink + "/" + util.PathEscapeSegments(ctx.Repo.TreePath)
 
+	if ctx.Repo.TreePath == ".editorconfig" {
+		_, editorconfigErr := ctx.Repo.GetEditorconfig()
```
Comment (delvh) [inline routers/web/repo/view.go:378]: Hmm, this error will always be in English.
I don't think there is anything we can do about that?

**Review by delvh — APPROVED** [review]
In general, your approach is fine.
However, I'm wondering whether we should display the warning somewhere else as well.
With the current approach, you'll likely never see the warning.
After all, who views exactly the editorconfig file using the web UI?
Most likely not many.

File: routers/web/repo/view.go (hunk 1 above)
Comment (silverwind) [reply routers/web/repo/view.go:378]: Besides running third party errors through AI translation, no, I see no way 😉

Comment (delvh) [reply routers/web/repo/view.go:378]: Thought so.

**Comment by 6543** [comment]
.

**Pull request merged**